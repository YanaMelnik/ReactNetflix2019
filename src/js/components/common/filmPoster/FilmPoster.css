.film__poster {
    max-width: 20vw;
}
