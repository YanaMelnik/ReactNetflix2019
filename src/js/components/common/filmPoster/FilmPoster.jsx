import React, { Component } from 'react';
import PropTypes from 'prop-types';

class FilmPoster extends Component {
  render() {
    return (
      <img src={this.props.imgUrl} alt={this.props.filmName}/>
    );
  }
}

FilmPoster.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  filmName: PropTypes.string.isRequired,

};

export default FilmPoster;
