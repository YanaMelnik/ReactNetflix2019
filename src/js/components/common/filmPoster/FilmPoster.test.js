import React from 'react';
import { shallow } from 'enzyme';
import FilmPoster from './FilmPoster';

describe('FilmPoster', () => {
    it('should render component correctly in case when all props defined', () => {
        const component = shallow(<FilmPoster imgUrl='url' filmName='name'/>);

        expect(component).toMatchSnapshot();
    });

    it('should show alternative text if we have some trouble with img url', () => {
        const component = shallow(<FilmPoster imgUrl='url' filmName='name'/>);

        expect(component.find('img').prop('alt')).toBe('name');
    });

    it('should show picture from img url', () => {
        const component = shallow(<FilmPoster imgUrl='url' filmName='name'/>);

        expect(component.find('img').prop('src')).toBe('url');
    });

});