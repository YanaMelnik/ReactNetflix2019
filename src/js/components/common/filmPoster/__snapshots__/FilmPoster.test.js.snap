// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPoster should render component correctly in case when all props defined 1`] = `
<img
  alt="name"
  className="film__poster"
  src="url"
/>
`;
